### PR TITLE
Remove location from exponential distribution

### PR DESCRIPTION
**Pull request by gordoncaleb**
The location of a distribution can controlled by just adding an offset. This PR removes the location hyper parameter in the ExponentialVertex that has been confusing people. Mathematically the location is now at 0.0.

### CODE REVIEW

File: keanu-project/src/main/java/io/improbable/keanu/vertices/dbl/probabilistic/ExponentialVertex.java [hunk 1]
```diff
@@ -33,69 +30,53 @@
...
         setValue(DoubleTensor.placeHolder(tensorShape));
     }
 
     /**
      * One to one constructor for mapping some shape of location and lambda to
      * location matching shaped exponential.
      *
-     * @param location the location of the Exponential with either the same shape as specified for this vertex or location scalar
-     * @param lambda   the lambda of the Exponential with either the same shape as specified for this vertex or location scalar
+     * @param lambda the lambda of the Exponential with either the same shape as specified for this vertex or location scalar
```
Comment (GeorgeNash) [inline keanu-project/src/main/java/io/improbable/keanu/vertices/dbl/probabilistic/ExponentialVertex.java]: Delete the `location` mention

File: keanu-project/src/main/java/io/improbable/keanu/vertices/dbl/probabilistic/ExponentialVertex.java [hunk 2]
```diff
@@ -33,69 +30,53 @@
...
 
-        this.location = location;
         this.lambda = lambda;
-        setParents(location, lambda);
+        setParents(lambda);
         setValue(DoubleTensor.placeHolder(tensorShape));
     }
 
     /**
      * One to one constructor for mapping some shape of location and lambda to
```
Comment (GeorgeNash) [inline keanu-project/src/main/java/io/improbable/keanu/vertices/dbl/probabilistic/ExponentialVertex.java]: Cheeky `location` still floating around.

File: keanu-project/src/test/java/io/improbable/keanu/distributions/gradient/Exponential.java [hunk 3]
```diff
@@ -14,20 +12,17 @@
...
-        double dPdx = -dPda;
-        return new Diff(dPda, dPdb, dPdx);
+    public static Diff dlnPdf(double b, double x) {
+        double dPdb = -(b - x) / Math.pow(b, 2);
+        double dPdx = -(1 / b);
+        return new Diff(dPdb, dPdx);
     }
 
     public static class Diff {
-        public final double dPda;
```
Comment (migwellian) [inline keanu-project/src/test/java/io/improbable/keanu/distributions/gradient/Exponential.java]: rename b to lambda?

File: keanu-project/src/main/java/io/improbable/keanu/vertices/dbl/probabilistic/ExponentialVertex.java (hunk 1 above)
Comment (gordoncaleb) [reply keanu-project/src/main/java/io/improbable/keanu/vertices/dbl/probabilistic/ExponentialVertex.java]: Nice catch

File: keanu-project/src/main/java/io/improbable/keanu/vertices/dbl/probabilistic/ExponentialVertex.java (hunk 2 above)
Comment (gordoncaleb) [reply keanu-project/src/main/java/io/improbable/keanu/vertices/dbl/probabilistic/ExponentialVertex.java]: Nice catch

File: keanu-project/src/test/java/io/improbable/keanu/distributions/gradient/Exponential.java (hunk 3 above)
Comment (gordoncaleb) [reply keanu-project/src/test/java/io/improbable/keanu/distributions/gradient/Exponential.java]: yup good catch

**Pull request merged**